File: 189-rotate-array/rotate-array.py
def reverse_rage(arr , low , high):
    while(low < high ):
        arr[high] , arr[low] = arr[low] , arr[high]
        high -=1
        low +=1



class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        rotation_comp = k%len(nums)
        reverse_rage(nums , 0 , len(nums)-1)
        reverse_rage(nums , 0 , rotation_comp-1)
        reverse_rage(nums , rotation_comp , len(nums)-1)
        return(nums)



        

File: 189-rotate-array/rotate-array.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
       	int rotation = k % nums.Length;
			List<int> Result = new List<int>();
			if (rotation == 0)
			{
				Console.Write(nums);
			}
			else
			{

				for (int i = (nums.Length - rotation); i < nums.Length; i++)
				{
					Result.Add(nums[i]);
					
				}

				for (int i = 0; i < (nums.Length - rotation) ; i++)
				{
					Result.Add(nums[i]);
				}


				for (int i = 0; i < nums.Length; i++)
				{
					nums[i] = Result[i];
					// Console.WriteLine(nums[i]);
				
				}
			}




    }
}